packages/elements/src/internals/machines/form/form.types.ts

```diff
@@ -18,10 +19,32 @@ export interface FeedbackOtherType extends FeedbackBase {
   message: string;
 }
 
-export type FieldDetails = {
+interface FieldDetailsBase {
   name?: string;
-  value?: string | readonly string[] | number;
   feedback?: FeedbackErrorType | FeedbackOtherType;
-};
+}
+
+export interface FieldDetailsWithValue extends FieldDetailsBase {
+  type:
+    | 'text'
+    | 'email'
+    | 'password'
+    | 'tel'
+    | 'number'
+    | 'date'
+    | 'time'
+    | 'datetime-local'
+    | 'month'
+    | 'week'
+    | 'otp';
```

Comment:
I thought i'd be able to make use of the `ClerkInputType` type here excluding checkbox, but that surfaced a bunch of type errors in `FIELD.ADD` and `FIELD.UPDATE`. Open to suggestions here.

```
type: Exclude<ClerkInputType, 'checkbox'>
```

Reply:
Honestly, I'm totally fine with this being explicit given we probably don't really want to support the entire set of `HTMLInputTypeAttribute` types.